src/ctx.cpp

```diff
@@ -3754,15 +3774,16 @@ llvm::Value *FunctionEmitContext::addVaryingOffsetsIfNeeded(llvm::Value *ptr, co
     bool is32bits = g->target->is32Bit() || g->opt.force32BitAddressing;
     llvm::Value *varyingOffsets = ProgramIndexVector(is32bits);
 
-    llvm::Value *offset = BinaryOperator(llvm::Instruction::Mul, unifSize, varyingOffsets);
+    llvm::Value *offset = BinaryOperator(llvm::Instruction::Mul, unifSize, varyingOffsets,
+                                         !baseType->GetAsUniformType()->IsUnsignedType());
```

Comment:
1. Why getting a uniform type is needed here?
2. Seems that direct checking if the type signed is needed.